ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+            {
+                Log.Error($"BenzingaDataConverter.Convert(): Data not found for date {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            WriteToFile(Process(newsArticles), date);
+            return true;
+        }
+
+        /// <summary>
+        /// Process the data to BenzingaNews instances
+        /// </summary>
+        /// <param name="dateDirectory">Directory for a given date, e.g. ./root/2018/12/27/</param>
+        /// <returns></returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
+            {
+                var document = new XmlDocument();
+                document.LoadXml(File.ReadAllText(publication.FullName));
+
+                var item = JsonConvert.DeserializeObject<JObject>(JsonConvert.SerializeXmlNode(document))["rss"]["channel"]["item"];
+                // Benzinga complies with RSS standards and uses RFC 822 for DateTime
+                var publicationDate = Parse.DateTime(item["pubDate"].ToString());
+
+                var tickers = new List<JToken>();
+
+                // Only process articles that contain tickers to disk
+                if (item["bz:ticker"] == null)
+                {
+                    continue;
+                }
+
+                // Check for only a single ticker before we try to iterate
+                if (item["bz:ticker"].Type == JTokenType.Object)
+                {
+                    tickers.Add(item["bz:ticker"]);
+                }
+                else
+                {
+                    foreach (var tickerData in JArray.Parse(item["bz:ticker"].ToString()))
+                    {
+                        tickers.Add(tickerData);
+                    }
+                }
+
+                var instance = JsonConvert.DeserializeObject<BenzingaNews>(item.ToString());
+
+                // Strip all HTML tags from the article, then convert HTML entities to their string representation
+                // e.g. "<html><p>Apple&#39;s Earnings</p></html>" would become "Apple's Earnings"
+                instance.Contents = WebUtility.HtmlDecode(Regex.Replace(instance.Contents, @"<[^>]*>", " "));
+                instance.Symbols = new List<BenzingaSymbolData>();
+                instance.Metadata = new BenzingaMetadata()
+                {
+                    IsPro = item["bz:type"]["@pro"].ToString() == "1",
+                    FirstRun = item["bz:type"]["@pro"].ToString() == "1",
+                    Kind = item["bz:type"]["#text"].ToString()
+                };
+
+                foreach (var ticker in tickers)
+                {
+                    var sentiment = ticker["@sentiment"] == null ? (decimal?)null : Parse.Decimal(ticker["@sentiment"].ToString());
+                    var exchange = ticker["@exchange"] == null ? string.Empty : ticker["@exchange"].ToString();
```

Comment:
If these were converted into objects above as suggested -- much of this code simplifies.